app/src/main/java/org/oppia/app/home/continueplaying/OngoingListAdapter.kt

```diff
@@ -0,0 +1,86 @@
+package org.oppia.app.home.continueplaying
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import org.oppia.app.databinding.OngoingStoryCardBinding
+import org.oppia.app.databinding.SectionTitleBinding
+
+private const val VIEW_TYPE_SECTION_TITLE_TEXT = 1
+private const val VIEW_TYPE_SECTION_STORY_ITEM = 2
+
+/** Adapter to inflate different items/views inside [RecyclerView]. The itemList consists of various ViewModels. */
+class OngoingListAdapter(
+  private val itemList: MutableList<ContinuePlayingViewModel>
+) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+    return when (viewType) {
+      // TODO(#216): Generalize this binding to make adding future items easier.
+      VIEW_TYPE_SECTION_TITLE_TEXT -> {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding =
+          SectionTitleBinding.inflate(
+            inflater,
+            parent,
+            /* attachToParent= */ false
+          )
+        SectionTitleViewHolder(binding)
+      }
+      VIEW_TYPE_SECTION_STORY_ITEM -> {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding =
+          OngoingStoryCardBinding.inflate(
+            inflater,
+            parent,
+            /* attachToParent= */ false
+          )
+        OngoingStoryViewHolder(binding)
+      }
+      else -> throw IllegalArgumentException("Invalid view type: $viewType")
+    }
+  }
+
+  override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+    when (holder.itemViewType) {
```

Comment:
Probably worth addressing the ``else`` case and throwing an exception in the invalid case.

Reply:
Done. added else case too.